### ISSUE TITLE
bug: Xcode build error "The specified item could not be found in the keychain"

### ISSUE THREAD

**Issue by sertal70** [ionic-team/capacitor#2588]
# Bug Report
Hi all
I started a new project based on Ionic5 for a mobile app, then added capacitor and opened Xcode to build it: everything goes well using simulator as target device, but an error occurs when I try to build it for a real iPhone attached on USB.
 
## Capacitor Version

`npx cap doctor` output:
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 1.5.1

  @capacitor/core: 1.5.1

  @capacitor/android: 1.5.1

  @capacitor/ios: 1.5.1

Installed Dependencies:

  @capacitor/cli 1.5.1

  @capacitor/core 1.5.1

  @capacitor/android 1.5.1

  @capacitor/ios 1.5.1

[success] Android looking great! 👌
  Found 0 Capacitor plugins for ios:
[success] iOS looking great! 👌

## Affected Platform(s)

- [ ] Android
- [x] iOS
- [ ] Electron
- [ ] Web

## Current Behavior

Running build in Xcode for real device I get the error _Build failed_ with the message `The specified item could not be found in the keychain`. I googled a lot about this error founding only issues related to certificate and keychain problems. Unfortunately none of those are my case because the certificate I'm using for signing the build is the same used without any issues in another Xcode project targeting the previous version of the same mobile app (Ionic 3 + Cordova based).
Having a look at the output messages in Xcode (see all log at the end of this post) one can see the commands generating the error:
```
...
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
error: The specified item could not be found in the keychain.
...
```
More strange than ever, if I copy&paste the codesign command in a terminal, it executes without any problems:
```shell
/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib: replacing existing signature
/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib: signed Mach-O universal (armv7 armv7s arm64 arm64e) [com.apple.dt.runtime.swiftos]
```

## Expected Behavior

I would expect the build process to complete without any errors

## Other Technical Details

`npm --version` output:
6.13.7

`node --version` output:
v12.14.1

`pod --version` output (iOS issues only):
1.9.1

## Other Information

Complete build output message list:
```

Showing Recent Messages
CopySwiftLibs /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app (in target 'App' from project 'App')
    cd /Volumes/Lavoro/Projects/sot-mobile/ios/App
    export CODESIGN_ALLOCATE=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate
    export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk
    builtin-swiftStdLibTool --copy --verbose --sign <my-signing-code-here> --scan-executable /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/App --scan-folder /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks --scan-folder /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/PlugIns --scan-folder /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/Pods_App.framework --platform iphoneos --toolchain /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain --destination /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks --strip-bitcode --strip-bitcode-tool /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip --emit-dependency-info /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Intermediates.noindex/App.build/Debug-iphoneos/App.build/SwiftStdLibToolInputDependencies.dep

Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftUIKit.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftUIKit.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreFoundation.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreFoundation.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftObjectiveC.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftObjectiveC.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftMetal.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftMetal.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftDarwin.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftDarwin.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftFoundation.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftFoundation.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftDispatch.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftDispatch.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftQuartzCore.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftQuartzCore.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreGraphics.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreGraphics.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftos.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftos.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreImage.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreImage.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCoreImage.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreImage.dylib
Copying /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCore.dylib to /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCore.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos/libswiftCore.dylib -r -o /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCore.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib: code object is not signed at all

Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftUIKit.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib: code object is not signed at all

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib: code object is not signed at all

Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftFoundation.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreFoundation.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDispatch.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftObjectiveC.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftDarwin.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftQuartzCore.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftCoreGraphics.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftMetal.dylib
Probing signature of /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
/usr/bin/codesign -r- --display /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
error: The specified item could not be found in the keychain.

error: The specified item could not be found in the keychain.

error: The specified item could not be found in the keychain.

error: The specified item could not be found in the keychain.

/Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib: code object is not signed at all

Codesigning /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
/usr/bin/codesign --force --sign <my-signing-code-here> --verbose /Users/Sergio/Library/Developer/Xcode/DerivedData/App-avabgaxilgqumifzfxtnugizxbks/Build/Products/Debug-iphoneos/App.app/Frameworks/libswiftos.dylib
error: The specified item could not be found in the keychain.

error: The specified item could not be found in the keychain.

error: Failed with exit code 1 (in target 'App' from project 'App')
```

**Comment by jcesarmobile** [comment]
can you provide a sample app?

**Comment by sertal70** [comment]
Hi @jcesarmobile , thanks for having a look into this.

Providing a sample app wouldd not help a lot, because I've been able to reproduce the issue starting from scratch with the following steps:

1. create a new blank Ionic project:
`ionic start myApp blank`
2. add capacitor to the project
`ionic integrations enable capacitor`
3. init app name and appId
`ionic cap init yourAppName yourAppId`
3. build the project
`ionic build`
4. add iOS platform to the project
`ionic cap add ios`
5. go to the Apple Developer website, login and create an app Identifier having "yourAppId" as identifier
6. open the project on Xcode
`ionic cap open ios`
7. In Xcode open config page "Signing & Capabilities", check "Automatically managing signing" box and then select the right value for you in the "Team" dropdown
8. Select "Generic iOS Device" on the toolbar
9. Select "Build" from the "Product" menu
10. One or more error "The specified item could not be found in the keychain" is shown on the screen

**Comment by jcesarmobile** [comment]
Ok, I can't reproduce, it's some problem on your computer or your certificates, but can be anything, try asking on Apple forums.

**Comment by sertal70** [comment]
Hi @jcesarmobile, to completely exclude that the issue was somehow related to my certificates or my Mac I made another test: using the same workflow but replacing capacitor with cordova. To be more specific, I did the follwing:

1.  create a new blank Ionic project:
`ionic start myApp blank`
2. add cordova to the project
`ionic cordova platform add ios`
3. configure app name and app identifier in `config.xml`
4. remove and add iOS platform to have config.xml propagate in iOS platform project
`ionic cordova platform rm ios && ionic cordova platform add ios`
5. build the project
`ionic build`
6. go to the Apple Developer website, login and create an app Identifier having the same app identifier you put in `config.xml`
7. open the project `platform/ios/yourapp.xcworkspace`on Xcode
8. In Xcode open config page "Signing & Capabilities", check "Automatically managing signing" box and then select the right value for you in the "Team" dropdown
Select "Generic iOS Device" on the toolbar
Select "Build" from the "Product" menu
9. observe Xcode showing the "Build succeeded" popup

So it seems that the issue is in capacitor rather than in my Mac. Please keep the issue open and let me know how can I help you to investigate it further.

**Comment by jcesarmobile** [comment]
No, it's not a Capacitor issue, Capacitor doesn't handle the signing of your app, that's done by Xcode itself. So it's probably some misconfiguration in Xcode.
Cordova doesn't use Xcode to sign.

**Comment by sertal70** [comment]
Sorry @jcesarmobile but I don't understand your answer. Let me outline the facts we have in this issue:

- there are two tools that generate an Xcode project for building a mobile app starting from a Ionic project: one is Capacitor, the other is Cordova.
- the artifacts generated by Capacitor don't build in Xcode due to an error during the sign process
- the artifacts generated by Cordova build in Xcode without any error.
- the certificate used to complete the build is the same in both cases

That said, come back to your answer: we all already know that "Capacitor doesn't handle the signing" or that "Cordova doesn't use Xcode to sign". Nevertheless Capacitor generates the Xcode project, so _maybe_  the generation process is buggy leading to the error we are discussing. Does it make any sense?
